core/src/main/java/com/backblaze/b2/client/B2StorageClientWebifierImpl.java

```diff
@@ -599,27 +614,58 @@ private String makeGetFileInfoByNameUrl(B2AccountAuthorization accountAuth,
     private String makeDownloadByNameUrl(B2AccountAuthorization accountAuth,
                                          String bucketName,
                                          String fileName,
-                                         String b2ContentDisposition) {
-        String url = accountAuth.getDownloadUrl();
-        if (!url.endsWith("/")) {
-            url += "/";
+                                         B2DownloadByNameRequest request) {
+        final String downloadUrl = accountAuth.getDownloadUrl();
+        final StringBuilder uriBuilder = new StringBuilder(downloadUrl);
+
+        if (!downloadUrl.endsWith("/")) {
+            uriBuilder.append("/");
         }
-        url += "file/" + bucketName + "/" + percentEncode(fileName);
-        url += maybeB2ContentDisposition('?', b2ContentDisposition);
-        return url;
+
+        uriBuilder
+                .append("file/")
+                .append(bucketName)
+                .append("/")
+                .append(percentEncode(fileName));
+
+        if (request != null) {
+            int queryparameterCount = 0;
+
+            queryparameterCount = maybeAddQueryParamToUrl(uriBuilder, queryparameterCount, "b2ContentDisposition", request.getB2ContentDisposition());
```

Comment:
someday we might want to have an interface for request objects that has these header accessor methods so we can share the code that adds them to the urls...?